src/us/kbase/workspace/database/provenance/EventDate.java

```diff
@@ -107,14 +83,78 @@ private static String checkDate(final String protoDate) {
 						final YearMonth ym = YearMonth.parse(protoDate);
 						return ym.toString();
 					default:
-						final LocalDate ymd = LocalDate.parse(protoDate, DateTimeFormatter.ISO_LOCAL_DATE);
+						final LocalDate ymd = LocalDate.parse(protoDate,
+								DateTimeFormatter.ISO_LOCAL_DATE);
 						return ymd.toString();
 				}
-			}
-			catch (DateTimeParseException e) {
+			} catch (DateTimeParseException e) {
 				// report the error below as an IllegalArgumentException
 			}
 		}
-		throw new IllegalArgumentException("Invalid date: \"" + protoDate + "\"\ndate must be in the format yyyy, yyyy-MM, or yyyy-MM-dd and be a valid combination of day, month, and year.");
+		throw new IllegalArgumentException(
+				"Invalid date: \"" + protoDate
+						+ "\"\ndate must be in the format yyyy, yyyy-MM, or yyyy-MM-dd and be a valid combination of day, month, and year.");
+	}
+
+	/**
+	 * Builds an {@link EventDate}.
+	 *
+	 * @param date  the date when the event occurred,
+	 *              in the format yyyy-MM-dd, yyyy-MM, or yyyy.
+	 * @param event the event that occurred on that date, as an Event
+	 * @return the new {@link EventDate}.
+	 */
+	public static EventDate build(final String date, final Event event) {
+		return build(date, event, new ArrayList<>());
+	}
+
+	/**
+	 * Builds an {@link EventDate}.
+	 *
+	 * @param date  the date when the event occurred,
+	 *              in the format yyyy-MM-dd, yyyy-MM, or yyyy.
+	 * @param event the event that occurred on that date, as a string.
+	 *              See the {@link Event} class for valid string values.
+	 * @return the new {@link EventDate}.
+	 */
+	public static EventDate build(final String date, final String event) {
+		final List<String> errorList = new ArrayList<>();
+		Event eventObject = null;
+		try {
+			eventObject = Event.getEvent(event);
+		} catch (Exception e) {
+			errorList.add(e.getMessage());
+		}
+		return build(date, eventObject, errorList);
+	}
+
+	/**
+	 * Builds an {@link EventDate}.
+	 *
+	 * @param date      the date when the event occurred,
+	 *                  in the format yyyy-MM-dd, yyyy-MM, or yyyy.
+	 * @param event     the event that occurred on that date, as an Event
+	 * @param errorList list of error strings accumulated during the `build` method
+	 *
+	 * @return the new {@link EventDate}.
+	 */
+	private static EventDate build(final String date, final Event event, final List<String> errorList) {
+		if (event == null && errorList.isEmpty()) {
+			errorList.add("event cannot be null");
+		}
+
+		String protoDate = null;
+		try {
+			final String trimmedDate = Util.checkString(date, "date");
+			protoDate = checkDate(trimmedDate);
+		} catch (Exception e) {
+			errorList.add(e.getMessage());
+		}
+
+		if (errorList.isEmpty()) {
+			return new EventDate(protoDate, event);
+		}
+		throw new IllegalArgumentException(
+				"Errors in EventDate construction:\n" + String.join("\n", errorList));
```

Comment:
all construction errors are collected and thrown as as `IllegalArgumentException`